types/atom.js.flow

```diff
@@ -2055,6 +2056,10 @@ type atom$AutocompleteProvider = {
   ) => void,
 };
 
+type atom$AutocompleteWatchEditor = (
+  editor: atom$TextEditor, labels: Array<string>
+) => ?atom$Disposable;
+
```

Comment:
I was not aware of this api and looked into it. I think we should put this in another file because it doesnt look like core atom api. Maybe even just at the top of `autocomplete.js`. The api is part of [autocomplete plus](https://github.com/atom/autocomplete-plus#the-watcheditor-api). We might want to link to that because `WatchEditor` api there might have ambiguous meaning since it's another type of `watch`

Reply:
Good idea I like it in file.

Reply:
I gonna introduce another non-core atom apis in #1710. 
So I think we can just make a file like `atom-pkgs.js.flow` (or i don't know, `atom-pkg-services.js.flow` ?) and moves atomcomplete types into this, that might make things more clear ?

Reply:
I was thinking of creating a `hydrogen-services.js.flow` where all types follow the format `hs_<package>$<type name>` hs stands for hydrogen services, and that way we know we typed this. package so we know where it is coming from in case the type every changes. type name is pretty self explanatory. What do you guys think? @aviatesk @BenRussert

Reply:
Services should be the only place we every require types outside atom or electron so thats why I think we should label it that way